src/cn-batch-forms.service.js

```diff
@@ -157,6 +162,24 @@
       }
     }
 
+    function processDiff(schema) {
+      const updateSchema = schema.params.updateSchema;
+      const links = _.filter(schema.batchConfig.links, ls => _.includes(ls, updateSchema));
+      const hardLinks = _.filter(schema.batchConfig.hardLinks, ls => _.includes(ls, updateSchema));
+      processSchemaDiff.call(this, schema.diff.schema, _.flatten(links.concat(hardLinks)));
+    }
+
+    function processSchemaDiff(properties, links) {
+      const props = _.keys(properties);
+      _.forEach(props, (prop) => {
+        if (_.has(properties[prop], "properties")) {
+          processSchemaDiff.call(this, properties[prop].properties, links);
+        } else if (_.every(links, l => !_.includes(l, prop))) {
```

Comment:
I'm not sure if this `_.includes` also needs to be changed to `_.startsWith`. This would be a great place to add a test to make sure we get the desired results for different possible scenarios like the one I mentioned above, or for when processing an array schema